### PR TITLE
Simplify testing validation problems

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

File: subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidatePluginsIntegrationTest.groovy [hunk 1]
```diff
@@ -493,8 +494,25 @@ class ValidatePluginsIntegrationTest extends AbstractPluginValidationIntegration
 
         expect:
         assertValidationFailsWith([
-            warning(notCacheableWithoutReason { type('MyTask').noReasonOnTask().includeLink() }),
-            warning(notCacheableWithoutReason { type('MyTransformAction').noReasonOnArtifactTransform().includeLink() })
+            warning(normalizeDocumentationLink("""Type 'MyTask' must be annotated either with @CacheableTask or with @DisableCachingByDefault.
```
[the remark this answers is not in the document]
Comment (lptr) [reply subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidatePluginsIntegrationTest.groovy:497]: That's extending the scope here a bit...

[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidatePluginsIntegrationTest.groovy:497]: I don't understand how that extends the scope if the `warning` method always calls `normalizeDocumentationLink` on the message.

[the remark this answers is not in the document]
Comment (lptr) [reply subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidatePluginsIntegrationTest.groovy]: I'd have to also change the `error()` method to do the same, which is called in a number of other places where we don't need normalization.

In any case, I found a better solution.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by lptr** [comment]
@bot-gradle test and merge.

**Comment by lptr** [comment]
@bot-gradle test and merge.

**Comment by lptr** [comment]
@bot-gradle test and merge.

**Comment by lptr** [comment]
@bot-gradle test and merge.

**Comment by lptr** [comment]
@bot-gradle test and merge please.

**Pull request merged**